src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -271,21 +272,25 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
      * @return the left curly corresponding to the block to be checked
      */
     private static DetailAST findLeftCurly(DetailAST ast) {
+        final DetailAST parent = ast.getParent();
+        final int parentType = parent.getType();
         final DetailAST leftCurly;
-        final DetailAST slistAST = ast.findFirstToken(TokenTypes.SLIST);
-        if ((ast.getType() == TokenTypes.LITERAL_CASE
-                || ast.getType() == TokenTypes.LITERAL_DEFAULT)
-                && ast.getNextSibling() != null
-                && ast.getNextSibling().getFirstChild() != null
-                && ast.getNextSibling().getFirstChild().getType() == TokenTypes.SLIST) {
-            leftCurly = ast.getNextSibling().getFirstChild();
+
+        if (parentType == TokenTypes.SWITCH_RULE) {
+            leftCurly = parent.findFirstToken(TokenTypes.SLIST);
         }
-        else if (slistAST == null) {
-            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        else if (parentType == TokenTypes.CASE_GROUP) {
+            final DetailAST statementList = ast.getNextSibling();
+            leftCurly = Optional.ofNullable(statementList)
+                         .map(DetailAST::getFirstChild)
+                         .filter(node -> node.getType() == TokenTypes.SLIST)
+                         .orElse(null);
```

Comment:
I think this is not a great usage of Optional (defaulting to a null value); it would make more sense for this method to return an Optional, if anything. How can we improve this?

Reply:
done